pkg/validation/validate_cluster.go

```diff
@@ -96,6 +109,41 @@ func getRoleNode(node *v1.Node) string {
 	return role
 }
 
+func collectComponentStatus(client k8s_clientset.Interface) (bool, error) {
+	componentList, err := client.CoreV1().ComponentStatuses().List(v1.ListOptions{})
+	if err != nil {
+		fmt.Println("first error")
+		return false, err
+	}
+
+	fmt.Println("looop")
+	for _, component := range componentList.Items {
+		for _, condition := range component.Conditions {
+			if condition.Status != "True" {
+				fmt.Println("bad status")
+				return false, nil
+			}
+		}
+	}
+	return true, nil
+}
+
+func collectPodHealth(client k8s_clientset.Interface) (bool, error) {
+	pods, err := client.CoreV1().Pods("kube-system").List(v1.ListOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	for _, pod := range pods.Items {
+		for _, status := range pod.Status.ContainerStatuses {
+			if !status.Ready {
+				return false, nil
```

Comment:
Same as ComponentStatuses - Instead of returning immediately, I think we should print out the bad pod and continue looking for other bad pods?  Just knowing the cluster is bad won't help resolve it